data-prepper-plugins/geoip-prepper/src/main/java/com/amazon/dataprepper/plugins/prepper/geoip/provider/LocationData.java

```diff
@@ -1,83 +1,83 @@
 package com.amazon.dataprepper.plugins.prepper.geoip.provider;
 
+import com.fasterxml.jackson.annotation.JsonProperty;
+
 import java.util.HashMap;
 import java.util.Map;
 import java.util.Objects;
 
 public class LocationData {
+
+    private final String CITY_FIELD = "geo.city_name";
+    private final String COUNTRY_FIELD = "geo.country_name";
+    private final String CONTINENT_FIELD = "geo.continent_code";
+    private final String COUNTRY_ISO_FIELD = "geo.country_iso_code";
+    private final String POSTAL_FIELD = "geo.postal_code";
+    private final String REGION_FIELD = "geo.region_name";
+    private final String REGION_CODE_FIELD = "geo.region_code";
+    private final String TIMEZONE_FIELD = "geo.timezone";
+    private final String LOCATION_FIELD = "geo.location";
+    private final String LATITUDE_FIELD = "geo.location.lat";
+    private final String LONGITUDE_FIELD = "geo.location.lon";
     private final String ip;
+
+    @JsonProperty(COUNTRY_FIELD)
     private final String countryName;
-    private final String subdivisionName;
+    @JsonProperty(REGION_FIELD)
+    private final String regionName;
+    @JsonProperty(CITY_FIELD)
     private final String cityName;
-    private final Map<String, Object> location;
+    @JsonProperty(LOCATION_FIELD)
+    private final Map<String, Double> location;
+    @JsonProperty(LATITUDE_FIELD)
     private final Double latitude;
+    @JsonProperty(LONGITUDE_FIELD)
     private final Double longitude;
+    @JsonProperty(CONTINENT_FIELD)
     private final String continentCode;
+    @JsonProperty(COUNTRY_ISO_FIELD)
     private final String countryCode;
+    @JsonProperty(POSTAL_FIELD)
     private final String postalCode;
+    @JsonProperty(REGION_CODE_FIELD)
     private final String regionCode;
-    private final String timezone;
+    @JsonProperty(TIMEZONE_FIELD)
+    private final String timeZone;
 
-    public LocationData(Map<Fields, Object> fields) {
-        ip = (String) fields.get(Fields.IP);
-        countryName = (String) fields.get(Fields.COUNTRY_NAME);
-        subdivisionName = (String) fields.get(Fields.REGION_NAME);
-        cityName = (String) fields.get(Fields.CITY_NAME);
-        Double[] latlon = (Double[]) fields.get(Fields.LOCATION);
-        continentCode = (String) fields.get(Fields.CONTINENT_CODE);
-        countryCode = (String) fields.get(Fields.COUNTRY_CODE2);
-        postalCode = (String) fields.get(Fields.POSTAL_CODE);
-        regionCode = (String) fields.get(Fields.REGION_CODE);
-        timezone = (String) fields.get(Fields.TIMEZONE);
+    public LocationData(Map<GeoDataField, Object> fields) {
```

Comment:
+1 for reconstruct this using builder pattern. We should manifest the input args instead of assuming one has the knowledge of fields in the map for future maintenance. You could put essential args in the constructor(If none, just use no arg constructor) and use with* method for optional args. You could refer to other plugins in the repo.